src/main/java/org/truffleruby/core/cast/BooleanCastNode.java

```diff
@@ -67,31 +66,32 @@ protected boolean doFloat(double value) {
         return true;
     }
 
-    @Specialization
-    protected boolean doBasicObject(DynamicObject object) {
-        return true;
-    }
-
     @Specialization
     protected boolean doSymbol(RubySymbol object) {
         return true;
     }
 
-    @Specialization(guards = "isForeignObject(object)", limit = "getCacheLimit()")
+    @Specialization(limit = "getCacheLimit()")
     protected boolean doForeignObject(Object object,
```

Comment:
Mmh, but this Specialization will now accept any object, including a `boolean`, `int`, etc.
It's still semantically correct because `InteropLibrary` works on primitives too, but seems suboptimal for those cases.
I'll try to add `RubyLibrary#isForeignObject` and `RubyLibrary#isRubyDynamicObject`.

Reply:
Yeah, this is good point :+1: Are you mean add `isForeignObject` and `isRubyDynamicObject` guards ?

Reply:
Right, those guards already exist on RubyGuards, so it seems fine to use them in this PR. Could you switch to it and then keep 2 separate specializations as before? (sorry for the back and forth)

They're not very efficient, but I'll fix that separately by using RubyLibrary, or by finishing the migration so all Ruby objects inherit from `RubyDynamicObject` (in progress).

Reply:
No problem, done. 
The example above works as expected: 
```js
js_null = Polyglot.eval('js', 'null')
[Object.new, js_null].map { |x| x ? true : false }
# => [true, false]
```